pg_backup_api/pg_backup_api/logic/utility_controller.py

```diff
@@ -201,21 +223,20 @@ def server_operation(server_name: str) \
     """
     Handle ``GET``/``POST`` request to ``/servers/*server_name*/operations``.
 
-    Get a list of recovery operations for *server_name*, if a ``GET`` request,
-    or create a new recovery operation for *server_name*, if a ``POST``
-    request.
+    Get a list of operations for *server_name*, if a ``GET`` request, or create
+    a new operation for *server_name*, if a ``POST`` request.
 
-    :param server_name: name of the Barman server related to the recovery
-        operation(s).
+    :param server_name: name of the Barman server related to the operation(s).
 
     :return: the returned response varies:
 
         * If a successful ``GET`` request, then return a JSON response with
-          ``operations`` key containing a list of IDs of recovery operations
-          for Barman server *server_name*;
+          ``operations`` key containing a list of operations for Barman server
```

Comment:
@gonzalemario This is the change which will break repmgr isn't it? [This code here](https://github.com/EnterpriseDB/repmgr/blob/d3b1ff45b0626c144db4cd1fb1f3c727ea06b2f3/pgbackupapi.c#L45-L48)  is currently expecting a list of IDs and it's now going to get a list of dicts.

Reply:
Right! That's what I meant [here](https://github.com/EnterpriseDB/pg-backup-api/pull/83#discussion_r1303326117).

Reply:
It would break repmgr if it did use it. [That code](https://github.com/EnterpriseDB/repmgr/blob/d3b1ff45b0626c144db4cd1fb1f3c727ea06b2f3/pgbackupapi.c#L45-L48) is part of `receive_operations_cb` which is a callback when `get_operations_on_server` is triggered. The purpose of that function is to check the different tasks the server has received for a specific node. That was added and included in v5.4.0 because I knew it was going to be useful. Let me explain:

In [repmgr-action-standby.c](https://github.com/EnterpriseDB/repmgr/blob/d3b1ff45b0626c144db4cd1fb1f3c727ea06b2f3/repmgr-action-standby.c#L7789C49-L7789C49) we added `run_pg_backupapi` to the pool of functions to create different postgres standby. That function is only using 1) the creation of a new recovery and 2) the check of that previous task. That's why we're not breaking *current* repmgr's standby creation mode but it will break things if we expand features in the future.

I agree also on having a 2.0 version, and I also agree that to separate the general refactoring from the API changes would be great.

Reply:
Ok great - so:

1. We should patch `receive_operations_cb` in repmgr so that it is not left with code which will not work against pg-backup-api.
2. We do not actually need to make a new repmgr release in order to be compatible with pg-backup-api 2.0.0 - the existing repmgr will work just fine with 1.1.1 and the proposed 2.0.0.

In that case I am -1 on the [suggestion I made yesterday](https://github.com/EnterpriseDB/pg-backup-api/pull/83#pullrequestreview-1592293518) to make the change to the output format of the endpoint optional - we should just update the format rather than add additional complexity, given it is not needed in order to maintain repmgr compatibility.